samples/common/src/utils.rs

```diff
@@ -116,75 +118,76 @@ pub async fn discover_digital_twin_provider_using_ibeji(
     }
 }
 
-/// Use Chariott to discover the endpoint for the digital twin service.
+/// Use Chariott to discover a service.
 ///
 /// # Arguments
-/// * `chariott_url` - Chariott's URL.
-pub async fn discover_digital_twin_service_using_chariott(
-    chariott_url: &str,
+/// * `chariott_uri` - Chariott's URI.
+/// * `namespace` - The service's namespace.
+/// * `name` - The service's name.
+/// * `version` - The service's version.
+pub async fn discover_service_using_chariott(
+    chariott_uri: &str,
+    namespace: &str,
+    name: &str,
+    version: &str,
 ) -> Result<String, Status> {
-    let mut client = ChariottServiceClient::connect(chariott_url.to_string())
+    let mut client = ServiceRegistryClient::connect(chariott_uri.to_string())
         .await
         .map_err(|e| Status::internal(e.to_string()))?;
 
-    let request = Request::new(FulfillRequest {
-        namespace: constants::chariott::NAMESPACE_FOR_IBEJI.to_string(),
-        intent: Some(IntentMessage { intent: Some(IntentEnum::Discover(DiscoverIntent {})) }),
+    let request = Request::new(DiscoverRequest {
+        namespace: namespace.to_string(),
+        name: name.to_string(),
+        version: version.to_string(),
     });
 
-    // Get list of services at the requested namespace, if any.
-    let fulfillment_result: Option<Vec<discover_fulfillment::Service>> = client
-        .fulfill(request)
-        .await?
-        .into_inner()
-        .fulfillment
-        .and_then(|fulfillment_message| fulfillment_message.fulfillment)
-        .and_then(|fulfillment_enum| match fulfillment_enum {
-            FulfillmentEnum::Discover(discover) => Some(discover.services.into_iter().collect()),
-            _ => None,
-        });
-
-    // If we discovered one or more service, then return the URL for the first one that uses gRPC.
-    match fulfillment_result {
-        Some(services) =>
-            services.iter()
-                .find(|service| service.schema_kind == constants::chariott::SCHEMA_KIND_FOR_GRPC)
-                .map(|service| service.url.clone())
-                .ok_or_else(|| Status::not_found("Failed to discover the in-vehicle digital twin service's URL, as none of the services found had the '{constants::chariott::SCHEMA_KIND_FOR_GRPC}' schema kind")),
-        None => Err(Status::not_found("Failed to discover the in-vehicle digital twin service's URL, as it is not registered with Chariott"))
+    let response =
+        client.discover(request).await.map_err(|error| Status::internal(error.to_string()))?;
+    let response_inner = response.into_inner();
```

Comment:
Since you're not using the result of this statement I'd combine it with the following one. You can also omit the if statement later by combining this with `ok_or` and `?`:

```rust
let uri = response
    .into_inner()
    .service
    .ok_or(Status::not_found("error message"))?
    .uri;
```

Reply:
The codes has changed since this was suggested and it is no longer valid.  Leaving as is.

Reply:
the suggestions are still relevant to the new code:

```rust
let service = response
    .into_inner()
    .service
    .ok_or(Status::not_found("error message"))?;
```

Reply:
Fixed.